### ISSUE TITLE
service set doesn't render Operator filter in assign rule

### ISSUE THREAD

**Issue by carraroj** [Icinga/icingaweb2-module-director#1777]
## Expected Behavior
filter addition assign rules with operator (NOT / OR / AND) see screenshot
```
 assign where ("test" in host.vars.bla || host.vars.foo == "test")
```
![image](https://user-images.githubusercontent.com/25805027/52415884-8b569f80-2ae8-11e9-804f-3b6e90f3a566.png)

## Current Behavior
```
NOT - OR - bla contains test - foo = test 
should be : assign where ! ("test" in host.vars.bla || host.vars.foo == "test")
```
Possible add feature NOT CONTAINS as operator
sample:
```
assign where host.vars.somevars not contains "some vars"
```

## Steps to Reproduce (for bugs)
- create a ServiceSet in Director with assign rule
Add operator (OR/NOT) expression to the assign statement for more filter expression

## Your Environment

* Director version (System - About):1.6.0
* Icinga Web 2 version and modules (System - About): 2.6.2
* Icinga 2 version (`icinga2 --version`): 2.10.2
* Operating System and version: CentOS 7.4
* Webserver, PHP versions: rh-php71

**Comment by carraroj** [comment]
ref/NC/599875

**Comment by Thomas-Gelf** [comment]
Please let me know whether this small change would fix your issue:

```patch
--- a/library/Director/IcingaConfig/AssignRenderer.php
+++ b/library/Director/IcingaConfig/AssignRenderer.php
@@ -257,7 +257,7 @@ class AssignRenderer
         }
 
         if ($filter instanceof FilterNot) {
-            return implode(' && ', $parts);
+            return '!(' . implode(' && ', $parts) . ')';
         } else {
             return implode($op, $parts);
         }
```

In case it doesn't please show me a step by step example

**Comment by carraroj** [comment]
yes, your little change solved the this issue, thanks, does it get into a supported/1.6 branch?

**Comment by Thomas-Gelf** [comment]
Be careful with this patch, it addresses the fact that 'NOT` on root level is rendered incorrectly. All other `NOT` operators were already work fine - and this sample patch breaks them.

I'll add related unit-tests for both variants, push a correct fix and merge it to support/1.6. So if you want we could have a tagged v1.6.1 by Monday.

**Comment by Thomas-Gelf** [comment]
@carraroj: a fix has been merged into master, unit tests have been added for both NOT on root level and for nested filters containing NOT. Fix has been backported to support/1.6 and will be released with v1.6.1.

**Comment by Thomas-Gelf** [comment]
@carraroj: all tests are green, so this looks good to me. Please let me know whether it fixed your issue.
